src/app/article/article.component.ts

```diff
@@ -1,33 +1,82 @@
-import { Component, OnInit } from '@angular/core';
-import { ArticleModel } from '../models';
-import { ArticlesService } from '../services';
-import { ActivatedRoute } from '@angular/router';
+import {Component, OnInit, ViewChild, TemplateRef, ViewContainerRef, OnDestroy} from '@angular/core';
+import {ArticleModel} from '../models';
+import {ArticlesService} from '../services';
+import {ActivatedRoute, Router} from '@angular/router';
+import {Store} from "@ngrx/store";
+import {finalize, Observable, startWith, Subscription} from 'rxjs';
+import {getIsUser} from "../auth/+store/auth.selector";
 
 @Component({
     selector: 'app-article',
     templateUrl: './article.component.html',
     styleUrls: ['./article.component.less'],
 })
-export class ArticleComponent implements OnInit {
+export class ArticleComponent implements OnInit, OnDestroy {
     commentErrors;
     comments;
     article: ArticleModel;
-    canModify: boolean = false;
+    canModify$: Observable<boolean>;
+    isLoading: boolean = false;
+    errors = null;
+    subscription: Subscription = new Subscription();
+    @ViewChild('modal_confirm') modalConfirm: TemplateRef<any>;
+    @ViewChild('vc', {read: ViewContainerRef}) vc: ViewContainerRef;
+
+    backdrop: any;
 
     constructor(
         private articlesService: ArticlesService,
-        private route: ActivatedRoute
-    ) {}
+        private route: ActivatedRoute,
+        private router: Router,
+        private store: Store
+    ) {
+    }
 
     ngOnInit(): void {
         this.article = this.route.snapshot.data['article'].article;
+        this.canModify$ = this.store.select(getIsUser(this.article.author.username))
+            .pipe(startWith(false));
+    }
+
+    openModal() {
+        let view = this.modalConfirm.createEmbeddedView(null);
```

Comment:
Not blocker: Ideally in real project we should move the logic and #modal_confirm itself into some independent service / component.  For example https://ng.ant.design/components/modal/en

```
import { Component } from '@angular/core';

@Component({
  selector: 'nz-demo-modal-basic',
  template: `
    <button nz-button [nzType]="'primary'" (click)="showModal()"><span>Show Modal</span></button>
    <nz-modal [(nzVisible)]="isVisible" nzTitle="The first Modal" (nzOnCancel)="handleCancel()" (nzOnOk)="handleOk()">
      <ng-container *nzModalContent>
        <p>Content one</p>
        <p>Content two</p>
        <p>Content three</p>
      </ng-container>
    </nz-modal>
  `
})
export class NzDemoModalBasicComponent {
  isVisible = false;

  constructor() {}

  showModal(): void {
    this.isVisible = true;
  }

  handleOk(): void {
    console.log('Button ok clicked!');
    this.isVisible = false;
  }

  handleCancel(): void {
    console.log('Button cancel clicked!');
    this.isVisible = false;
  }
}
```